Feature/tariik mvp/development home

TarikVelic opened the pull request:
Development page. 

--------------------------------------------------------------------
 MAIN
* Development Home Page
* Duration style in modal
--------------------------------------------------------------------
 OTHER
 * ////
--------------------------------------------------------------------
![Screenshot_6](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/101573891/0af0c155-fb0b-408e-819f-f07a809d54c7)
![Screenshot_7](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/101573891/f9742086-f2de-4d32-8547-562be24a80dd)

![Screenshot_3](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/101573891/3fb6e9ea-388f-447c-a9b0-cd90cabbc12b)

TarikVelic committed:
Modal duration style

--------------------------------------------------------------------
 MAIN
 * Changed width for projects table
 * Added style for duration from figma
 * Added new FormField for Date
 * Small code fixes
 * REBASE
--------------------------------------------------------------------
 OTHER
 *
--------------------------------------------------------------------

TarikVelic committed:
Modal duration style

--------------------------------------------------------------------
 MAIN
 * Changed width for projects table
 * Added style for duration from figma
 * Added new FormField for Date
 * Small code fixes
--------------------------------------------------------------------
 OTHER
 *
--------------------------------------------------------------------

TarikVelic committed:
color change

TarikVelic committed:
color change

TarikVelic committed:
small changes code

TarikVelic committed:
small changes code

TarikVelic committed:
FEATURE/TariikMVP/DevelopmentHome

--------------------------------------------------------------------
 MAIN
 * Development page in Home WORK IN PROGRESS
--------------------------------------------------------------------
 OTHER
 * <other msg>
--------------------------------------------------------------------

TarikVelic committed:
FEATURE/TariikMVP/DevelopmentHome

--------------------------------------------------------------------
 MAIN
 * Development page in Home WORK IN PROGRESS
--------------------------------------------------------------------
 OTHER
 * <other msg>
--------------------------------------------------------------------

ElvaRivle committed:
ElvaRivle/Misc/BugFixes (#27)

* ElvaRivle/Misc/BugFixes

--------------------------------------------------------------------
 MAIN
 * fixed all bugs that we discussed last time
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * logout functionality

* ElvaRivle/Misc/BugFixes

--------------------------------------------------------------------
 MAIN
 * fixed all bugs that we discussed last time
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * ~~logout functionality~~

* ElvaRivle/Misc/BugFixes

--------------------------------------------------------------------
 MAIN
 * .env.example added
 * typo in variable namings
 * API_URL again loaded from .env
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

* ElvaRivle/Misc/BugFixes

--------------------------------------------------------------------
 MAIN
 * project modal fixed action names
 * fixed PR review
--------------------------------------------------------------------
 OTHER
 * /

---------

Co-authored-by: Elvir Vlahovljak <evlahovlja1@etf.unsa.ba>

TarikVelic committed:
Development Charts WIP

TarikVelic committed:
Development Charts WIP

TarikVelic committed:
Development Home Charts WIP

TarikVelic committed:
Development Home Charts WIP

TarikVelic committed:
Development Home WIP

TarikVelic committed:
Development Home WIP

becarselman commented on src/components/home/devRevenueCosts/Development.js:
```diff
@@ -0,0 +1,170 @@
+          {...RevenueCostsMonth}
+          data={project}
+          yAxis={
+            index === 1 || index === 2
```
why you use 2 conditions when you can use index !== 0 that will cover 1 and 2

becarselman commented on src/components/home/devRevenueCosts/Development.js:
```diff
@@ -0,0 +1,170 @@
+              </a>
+              </div>
+              <Row gutter={[16, 16]}>
+  {RevenueCostsM.map((project, index) => (
```
move map outside of jsx

becarselman commented on src/components/home/devRevenueCosts/Development.js:
```diff
@@ -0,0 +1,170 @@
+</div>
+          <div>
+          <Row gutter={[16, 16]}>
+  {RevenueGap.map((project) => (
```
move map outisde of jsx

becarselman commented on src/components/home/devRevenueCosts/Development.scss:
```diff
@@ -0,0 +1,285 @@
+    cursor: pointer;
+}
+
+.selection1{
```
you have almost same style for all 3 sections. can we use just one

TarikVelic replied:
I just copy paste Home.scss, but I will correct it

TarikVelic replied:
I copied Home.scss, but I will correct it

becarselman commented on src/components/home/devRevenueCosts/Development.scss:
```diff
@@ -0,0 +1,285 @@
+    text-align: center;
+}
+
+.selection1.highlight,
```
same why you don't use only one section

becarselman commented:
can we get new screenshots

TarikVelic commented:
![Screenshot_1](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/101573891/a89d34c0-49ae-466c-a9b7-4f0fa4820a34)

New screenshot

TarikVelic committed:
Fixed review comments

TarikVelic committed:
Fixed review comments

becarselman commented on src/components/home/devRevenueCosts/Options.js:
```diff
@@ -0,0 +1,74 @@
+  maxColumnWidth: 20,
+  dodgePadding: 6,
+  color: ({ type }) => {
+    if (type === 'Grand Total Planned Revenue') {
```
this is good for hardcoded data. But if we change something this will not work at all.

becarselman commented on src/components/formFields/FormField.scss:
```diff
@@ -117,4 +117,51 @@ button[data-style="secondary"] {
+.date-range-input {
+  padding: 8px 43px;
+  gap: 16px; 
+  font-family: "Gilroy", sans-serif;
```
I think we have mixin for this

becarselman commented on src/components/formFields/FormField.scss:
```diff
@@ -117,4 +117,51 @@ button[data-style="secondary"] {
+}
+
+.ant-picker .ant-picker-input {
+  display: block ruby !important;
```
avoid using important

becarselman commented on src/images/home/square.svg:
This can not be image. Please code this

TarikVelic committed:
Fixed review comments in Development home page

TarikVelic committed:
Fixed review comments in Development home page

ElvaRivle committed:
ElvaRivle/Feature/kzirDj9o (#33)

* ElvaRivle/Feature/kzirDj9o

--------------------------------------------------------------------
 MAIN
 * when employee clicked, modal correctly populated
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * correct modal look or create new edit modal (heading, bottom buttons)
 * add hidden _id field
 * test with backend
 * implement for Projects
 *

* ElvaRivle/Feature/kzirDj9o

--------------------------------------------------------------------
 MAIN
 * edit employee modal done
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * ~~correct modal look or create new edit modal (heading, bottom buttons)~~
 * ~~add hidden _id field~~
 * ~~test with backend~~
 * implement for Projects

* ElvaRivle/Feature/kzirDj9o

--------------------------------------------------------------------
 MAIN
 * when project clicked, modal correctly populated, except for employees
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * ~~correct modal look or create new edit modal (heading, bottom buttons)~~
 * ~~add hidden _id field~~
 * ~~test with backend~~
 * ~~implement for Projects~~
 * implement PUT route for projects
 * find a way to select proper employees when clicked on project

becarselman commented on src/components/formFields/FormField.scss:
```diff
@@ -117,4 +117,48 @@ button[data-style="secondary"] {
+  display: block ruby;
+}
+
+.ant-picker .ant-picker-input > input {
```
This will be done on whole project scope and we don't want to do that. For example someone want to do something similar your style will block him in executing this. Please use custom classNames and than you can do this inside

becarselman commented on src/components/home/devRevenueCosts/Options.js:
```diff
@@ -0,0 +1,74 @@
+  maxColumnWidth: 20,
+  dodgePadding: 6,
+  color: ({ type }) => {
+    const colorMap = {
```
this is good for demo purpose but we will expect live data and there will not be Grand Total Planned Revenue project and than this colors will not work. You need to figure out some different approach to do this. Myb you can create array of 10 colors and than randomly pick color for each project and than set al aver places.

becarselman replied:
To do that you will need to upadate this RevenueCostsMonth to be function that will take parameter color.

TarikVelic committed:
Development update

TarikVelic committed:
Development update

TarikVelic committed:
Scss change

TarikVelic committed:
Scss change

TarikVelic committed:
Merge branch 'FEATURE/TariikMVP/DevelopmentHome' of https://github.com/becarselman/ant-colony-pharaoh-frontend into FEATURE/TariikMVP/DevelopmentHome

TarikVelic committed:
Rebase with develop

TarikVelic committed:
For demo purpose

ElvaRivle committed:
ElvaRivle/Misc/BugFixes-V2 (#36)

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * notifications added on employees modals
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * close modal on success, don't close on error
 * fetch collection (projects, employees...) after modal closed with success
 * scrollable modal

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * notifications added on employees modals
 * upon successful add employee, modal closed and table refreshed
--------------------------------------------------------------------
 OTHER
 * fixed very annoying bug that caused page variable to be undefined
--------------------------------------------------------------------
 TODO
 * ~~close modal on success, don't close on error~~
 * ~~fetch collection (projects, employees...) after modal closed with success~~
 * do the same for other modals and projects table
 * scrollable modal
 * Edit Projects modal doesn't work, developers field is the issue

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * notifications added on employees modals
 * upon successful add/edit employee, modal closed and table refreshed
--------------------------------------------------------------------
 OTHER
 * fixed very annoying bug that caused page variable to be undefined
--------------------------------------------------------------------
 TODO
 * ~~close modal on success, don't close on error~~
 * ~~fetch collection (projects, employees...) after modal closed with success~~
 * do the same for other modals and projects table
 * scrollable modal
 * Edit Projects modal doesn't work, developers field is the issue

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * notifications added on employees modals
 * upon successful add/edit employee, modal closed and table refreshed
 * same thing for projects
--------------------------------------------------------------------
 OTHER
 * fixed very annoying bug that caused page variable to be undefined
 * moved modalActive state from component to redux
 * projectStatus moved to redux store
--------------------------------------------------------------------
 TODO
 * ~~close modal on success, don't close on error~~
 * ~~fetch collection (projects, employees...) after modal closed with success~~
 * ~~do the same for other modals and projects table~~
 * scrollable modal
 * Edit Projects modal doesn't work, developers field is the issue

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * notifications added on employees and projects modals
 * upon successful add/edit employee, modal closed and table refreshed
 * same thing for projects
 * when modal is opened, body scroll disabled, modal scroll  enabled
--------------------------------------------------------------------
 OTHER
 * fixed very annoying bug that caused page variable to be undefined
 * moved modalActive state from component to redux
 * projectStatus moved to redux store
--------------------------------------------------------------------
 TODO
 * ~~close modal on success, don't close on error~~
 * ~~fetch collection (projects, employees...) after modal closed with success~~
 * ~~do the same for other modals and projects table~~
 * ~~scrollable modal~~
 * Edit Projects modal doesn't work, developers field is the issue

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * wrong div has overflow
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * wrong div has overflow
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

* ElvaRivle/Misc/BugFixes-V2

--------------------------------------------------------------------
 MAIN
 * notifications added on employees and projects modals
 * upon successful add/edit employee, modal closed and table refreshed
 * same thing for projects
 * when modal is opened, body scroll disabled, modal scroll enabled
 * Edit Projects modal works
--------------------------------------------------------------------
 OTHER
 * fixed very annoying bug that caused page variable to be undefined
 * moved modalActive state from component to redux
 * projectStatus moved to redux store
--------------------------------------------------------------------
 TODO
 * ~~close modal on success, don't close on error~~
 * ~~fetch collection (projects, employees...) after modal closed with success~~
 * ~~do the same for other modals and projects table~~
 * ~~scrollable modal~~
 * ~~Edit Projects modal doesn't work, developers field is the issue~~

ElvaRivle committed:
Revert "ElvaRivle/Misc/BugFixes-V2 (#36)"

This reverts commit ca5ffc67bba76d6b6ae737c67eab71fecfd998e9.

ElvaRivle committed:
Merge pull request #37 from becarselman/revert-36-ElvaRivle/Misc/BugFixes-V2

Revert "ElvaRivle/Misc/BugFixes-V2"

TarikVelic committed:
Development review comment fix

TarikVelic committed:
Merge branch 'FEATURE/TariikMVP/DevelopmentHome' of https://github.com/becarselman/ant-colony-pharaoh-frontend into FEATURE/TariikMVP/DevelopmentHome

TarikVelic commented:
![Screenshot_5](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/101573891/cd79911f-0e74-4dd2-9eb3-652e377fcf16)
